### PR TITLE
fix: move metastore db credentials into a secret

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Fixes: https://github.com/stackabletech/hive-operator/issues/148
Jenkins :green_circle: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hive-operator-it-custom/56/

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -830,6 +831,22 @@ fn build_metastore_rolegroup_statefulset(
...
+        DB_USERNAME_ENV,
+        &credentials_secret_name,
+        "username",
+    )];
+    env.push(env_var_from_secret(
+        DB_PASSWORD_ENV,
+        &credentials_secret_name,
+        "password",
+    ));
+    container_builder.add_env_vars(env);
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:848]:
```suggestion
    container_builder.add_env_vars(vec![
        env_var_from_secret(DB_USERNAME_ENV, &credentials_secret_name, "username"),
        env_var_from_secret(DB_PASSWORD_ENV, &credentials_secret_name, "password"),
    ]);
```

File: tests/templates/kuttl/cluster-operation/10-install-hive.yaml.j2 [hunk 2]
```diff
@@ -15,9 +15,8 @@ spec:
   clusterConfig:
     database:
       connString: jdbc:derby:;databaseName=/tmp/hive;create=true
-      user: APP
-      password: mine
       dbType: derby
+      credentialsSecret: hive-credentials
```
Comment (maltesander) [inline tests/templates/kuttl/cluster-operation/10-install-hive.yaml.j2:19]: Can we make this consistent (i would prefer having `credentialSecret` below `connString` as in other examples)?

File: tests/templates/kuttl/smoke/60-install-hive.yaml.j2 [hunk 3]
```diff
@@ -15,9 +15,8 @@ spec:
   clusterConfig:
     database:
       connString: jdbc:postgresql://postgresql:5432/hive
-      user: hive
-      password: hive
       dbType: postgres
+      credentialsSecret: hive-credentials
```
Comment (maltesander) [inline tests/templates/kuttl/smoke/60-install-hive.yaml.j2:19]: Lets move `credentialsSecret` up as in other examples/tests?

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,10 @@ All notable changes to this project will be documented in this file.
 
 - Added documentation/tutorial on using external database drivers ([#449]).
 
+### Fixed
+
+- [BREAKING] Move the metastore DB credentials out fo the CRD into a secret ([#452]).
```
Comment (maltesander) [inline CHANGELOG.md:13]:
```suggestion
- [BREAKING] Move the metastore `user` and `password` DB credentials out of the CRD into a Secret containing the keys `username` and `password` ([#452]).
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Just nitpicking, LGTM otherwise!

File: tests/templates/kuttl/smoke/60-install-hive.yaml.j2 (hunk 3 above)
Comment (adwk67) [reply tests/templates/kuttl/smoke/60-install-hive.yaml.j2]: Done: 67ed2e0

File: tests/templates/kuttl/cluster-operation/10-install-hive.yaml.j2 (hunk 2 above)
Comment (adwk67) [reply tests/templates/kuttl/cluster-operation/10-install-hive.yaml.j2]: Done: 67ed2e0

**Pull request merged**